rust/core-lib/src/view.rs

```diff
@@ -487,3 +490,23 @@ fn clamp(x: usize, min: usize, max: usize) -> usize {
         max
     }
 }
+
+/// Computes a new selection based on applying a delta to the old selection.
+///
+/// When new text is inserted at a caret, the new caret can be either before
+/// or after the inserted text, depending on the `after` parameter.
+// Should this move to Selection itself?
```

Comment:
Moving this into `Selection` itself, would make `selection.rs` less self-contained. However, it kind of feels out of place here, so that I would personally prefer it being in Selection itself.